File: 2302-count-subarrays-with-score-less-than-k/2302-count-subarrays-with-score-less-than-k.py
class Solution:
    def countSubarrays(self, A: List[int], k: int) -> int:
        n = len(A)
        res = 0
        acc = list(accumulate(A, initial = 0))
        l = 0
        for i in range(n):
            while (acc[i + 1] - acc[l])*(i - l + 1) >= k:
                l += 1
            res += i - l + 1
        return res

File: 2302-count-subarrays-with-score-less-than-k/2302-count-subarrays-with-score-less-than-k.java
class Solution {
    public long countSubarrays(int[] nums, long k) {
        long[] ps = new long[nums.length + 1];
        for(int i = 1; i <= nums.length; i++) {
            ps[i] = ps[i - 1] + nums[i - 1];
        }
        int l = 0, curr = 0;
        long cnt = 0;
        for(int i = 0; i < nums.length; i++) {
            while((ps[i + 1] - ps[l]) * (i - l + 1) >= k) {
                l++;
            }
            cnt += i - l + 1;
        }
        return cnt;
    }
}